exchanges/bitfinex/bitfinex_test.go

```diff
@@ -429,9 +442,37 @@ func TestUpdateTicker(t *testing.T) {
 }
 
 func TestUpdateTickers(t *testing.T) {
-	err := b.UpdateTickers(context.Background(), asset.Spot)
+	t.Parallel()
+
+	err := b.UpdateTradablePairs(context.Background(), false)
 	if err != nil {
-		t.Error(err)
+		t.Fatal(err)
+	}
+
+	assets := b.GetAssetTypes(false)
+	for x := range assets {
+		var avail []currency.Pair
+		avail, err = b.GetAvailablePairs(assets[x])
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		err = b.CurrencyPairs.StorePairs(assets[x], avail, true)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		err = b.UpdateTickers(context.Background(), assets[x])
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for y := range avail {
```

Comment:
@shazbert This turns out to be a little unstable;
When Bitfinex [delisted ICE](https://www.bitfinex.com/posts/990) it's still coming back in all pub:conf and pub:info listings:
```
❯ curl -s https://api.bitfinex.com/v2/conf/pub:info:pair | jq -c '.[0][] | select(.[0] | test("(BTC|ICE|ZIL)USD")) '                                                                   ["BTCUSD",[null,null,null,"0.00006","2000.0",null,null,null,0.1,0.05,null,null]]
["ICEUSD",[null,null,null,"4.0","25000.0",null,null,null,null,null,null,null]]
["ZILUSD",[null,null,null,"40.0","1500000.0",null,null,null,null,null,null,null]]
```
_( I included ZIL to show a tradable pair without Margin fields )_
There's absolutely no sign it's not a tradable pair _until_ you ask for a ticker for it:
```
❯ curl -s https://api.bitfinex.com/v2/ticker/tICEUSD                                                                                                                                   ["error",10020,"symbol: invalid"]⏎
❯ curl -s https://api.bitfinex.com/v2/ticker/tBTCUSD                                                                                                                                   [42854,11.8920918,42855,12.71095442,-290,-0.00672292,42846,725.08132142,43288,41850]⏎
```

So I suggest that we probably limit the checks to show that *a* ticker works for each asset.
Does that work for you?

Reply:
Works for me! Thanks.